### ISSUE TITLE
Iputils fails to install on Centos7 image

### ISSUE THREAD

**Issue by baspeeters** [CentOS/sig-cloud-instance-images#77]
Dockerfile:
```
FROM centos:centos7

RUN yum -y update && yum -y install iputils
```

Yum output:

```
...

  Updating   : dracut-033-463.el7.x86_64                                 68/159 
  Updating   : kmod-20-9.el7.x86_64                                      69/159 
  Updating   : systemd-219-30.el7_3.6.x86_64                             70/159 
/var/tmp/rpm-tmp.AtfB5w: line 1: /etc/sysconfig/network-scripts/network-functions: No such file or directory
warning: %triggerin(systemd-219-30.el7_3.6.x86_64) scriptlet failed, exit status 1
Non-fatal <unknown> scriptlet failure in rpm package systemd-219-30.el7_3.6.x86_64
  Updating   : 1:dbus-1.6.12-17.el7.x86_64                               71/159 
  Updating   : iputils-20160308-8.el7.x86_64                             72/159Error unpacking rpm package iputils-20160308-8.el7.x86_64
 
error: unpacking of archive failed on file /usr/bin/ping: cpio: cap_set_file
  Updating   : yum-plugin-ovl-1.1.31-40.el7.noarch                       73/159 
error: iputils-20160308-8.el7.x86_64: install failed
  Updating   : yum-utils-1.1.31-40.el7.noarch                            74/159 
  Updating   : glib2-2.46.2-4.el7.x86_64                                 75/159

...

Failed:
  iputils.x86_64 0:20121221-7.el7        iputils.x86_64 0:20160308-8.el7
```

**Comment by jperrin** [comment]
what host are you running this on?

**Comment by FlorianHeigl** [comment]
I just ran into the same on a Scaleway VM, the type is VC1M, kernel is: 
Linux scw-7688cf 4.10.8-docker-1 #1 SMP Tue Apr 4 16:18:07 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
That's a VM.
I retried on a bare metal system there, same thing.


IMO it'll be the same on any locked down docker kernel, if it works it's either _super fine grained, well configured_ (unlikely in real world) or simply not secured (more likely imo)

**Comment by FlorianHeigl** [comment]
I just found there's two docker issues on this:
https://github.com/moby/moby/issues/5928
https://github.com/moby/moby/issues/6980

sounds quite kernel related. I'll follow up if i manage to work around.

**Comment by FlorianHeigl** [comment]
The common, go-to solution seems to be not fixing AUFS setups, but switching away from AUFS "storage" driver to i.e. device-mapper. IIRC that's the dangerous one that runs LVM thin pools on top of loop devices.
It didn't work in my case though, but THIS seems to be a very good explanation: http://www.projectatomic.io/blog/2015/04/problems-with-ping-in-containers-on-atomic-hosts/

I wish you all great fun in the world of easily reproducible immutable containers. :)

**Comment by jperrin** [comment]
Yup. You need to not use AUFS. Even the base docker containers for CentOS are built specifically excluding AUFS -> https://github.com/docker-library/official-images/blob/master/library/centos#L4

**Comment by FlorianHeigl** [comment]
A Centos container *image* would need to be able to run on an AUFS host anyway.
A _yum upgrade_ of the **iputils** package will mess around with the capabilities of ping, which is what's causing this issue.
Another fun bit is **selinux-policy-targeted** which tries to call out to dbus.
Maybe these things are excluded in the basic centos docker container and not in the one from scaleway. That would be an explanation.
If not, it's overeager %POST scripts of some RPMs which should be cleansed with fire.

**Comment by FlorianHeigl** [comment]
Last thing I can help with, this is how I managed to at least build my image...

You need to exclude those things to work around both issues.
```RUN yum --exclude=openssh-\* --exclude=policycoreutils\* --exclude=libsemanage-\* --exclude=selinux-\* --exclude=iputils update -y```

That of course does mean you would miss out on multiple security fixes which then need to be re-applied on first boot. Tracking the very most recent centos cloud image will surely help once the package in that image is updated.

**Comment by ValeryDubrava** [comment]
I have the same issue on the dedicated server (not a virtual environment).

**Comment by jperrin** [comment]
@ValeryDubrava I'd ask you to please either file a bug at bugs.centos.org with the full output of the yum command, or to ask on the mailing list. This repo (and associated issue tracker) is just for the container and vagrant pieces.
